### DISCUSSION TITLE
push: true not working on postgresAdapter

### DISCUSSION THREAD

**Discussion by woo94** [payloadcms/payload#7511] [Q&A]
### Link to reproduction

Restart the server

### Describe the Bug

push: true option on postgresAdapter is not reflected to the real database.

[02:01:35] ERROR (payload): error: column "nn" does not exist
    at /Users/woo94/Desktop/artistravel.io/my-duck-map-backend/node_modules/pg/lib/client.js:526:17
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/woo94/Desktop/artistravel.io/my-duck-map-backend/node_modules/src/node-postgres/session.ts:64:19
    at async find (/Users/woo94/Desktop/artistravel.io/my-duck-map-backend/node_modules/@payloadcms/db-postgres/src/find/findMany.ts:159:19)
    at async find (/Users/woo94/Desktop/artistravel.io/my-duck-map-backend/node_modules/payload/src/collections/operations/find.ts:145:16)
    at async findHandler (/Users/woo94/Desktop/artistravel.io/my-duck-map-backend/node_modules/payload/src/collections/requestHandlers/find.ts:30:20)

The phrase "development mode only" is very ambiguous. I thought I could enter development mode by adding NODE_ENV=development to the .env file, but that wasn't the case either.

https://payloadcms.com/docs/database/postgres#options doesn't help me at all

### To Reproduce

here is my code

```ts
// product.ts which is collection
import { nanoid } from "nanoid";
import { CollectionConfig } from "payload/types";

export const CMSCollection_Product: CollectionConfig = {
  slug: "product",
  fields: [
    {
      name: "id",
      type: "text",
      unique: true,
    },
    {
      name: "n",
      type: "number",
    },
    // {
    //   name: "nn",
    //   type: "number",
    // },
  ],
  hooks: {
    beforeChange: [
      async ({ data, operation, originalDoc }) => {
        // id를 적지 않고 product를 생성하는 경우
        if (operation == "create" && !data.id) {
          data.id = nanoid(16);
        }

        // duplicate를 하는 경우
        if (operation == "create" && originalDoc) {
          data.id = nanoid(16);
        }
        return data;
      },
    ],
  },
};
```

```ts
// payload.config.ts
import { buildConfig } from "payload/config";
import { slateEditor } from "@payloadcms/richtext-slate";
import { postgresAdapter } from "@payloadcms/db-postgres";
import { webpackBundler } from "@payloadcms/bundler-webpack";
import { CMSCollection_Product } from "./payload-cms/collections";

export default buildConfig({
  admin: {
    bundler: webpackBundler(),
  },
  editor: slateEditor({}),
  db: postgresAdapter({
    pool: {
      host: "127.0.0.1",
      port: 5432,
      user: process.env.DB_USER,
      password: process.env.DB_PASSWORD,
      database: process.env.DB_DATABASE,
    },
    schemaName: "celetrip-cms",
    push: true,
  }),
  collections: [CMSCollection_Product],
});
```

Restart the server doesn't change the table columns.

### Payload Version

2.25.0

### Adapters and Plugins

db-postgres, bundler-webpack

**Comment by jmikrut** [comment]
Hey @woo94 — `push: true` is the default setting, but it only works in development mode. You should not use it for production databases, so even though you can define it, it's not going to work when you start the server in production. You need to create a migration and then run the migration against your production database instead. 

Is this what you're referring to? The fact that `push: true` is not respected while running in production?

We can definitely update our docs to add more clarity though. I think that would be good.

**Reply by woo94** [reply]
@jmikrut how can i enter to development mode? 
My database is a development database hosted on Docker Desktop.

In a production environment, I am accustomed to writing migration scripts, but I do not want to do that during the development stage.

It's not because of "push: true". I couldn't find any method anywhere to automatically reflect collection field changes in the database. If there is a method, please let me know.
